From b0ec72e3651aeffca2e31849e6f8d264f933270e Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 19 Mar 2024 09:48:48 +0000
Subject: [PATCH] ARM: dts: pi400: Force stdout-path to serial0

The upstream Pi DTS files select serial1 for stdout-path, but because of
the way the downstream kernel numbers UARTS, serial0 is required in the
downstream DTS. Using the wrong value breaks U-boot.

In the 6.6 kernel the downstream Pi 400 DTS was made closer to upstream,
inheriting most of its content from the Pi 4B DTS, but because of the
order of inclusion it lost the override for stdout-path. Restore that
override.

See: https://github.com/raspberrypi/firmware/issues/1875
Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/bcm2711-rpi-400.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/broadcom/bcm2711-rpi-400.dts b/arch/arm/boot/dts/broadcom/bcm2711-rpi-400.dts
index 45c3676b3d16..d5b81b889018 100644
--- a/arch/arm/boot/dts/broadcom/bcm2711-rpi-400.dts
+++ b/arch/arm/boot/dts/broadcom/bcm2711-rpi-400.dts
@@ -43,6 +43,12 @@ &pm {
 // =============================================
 // Downstream rpi- changes
 
+/ {
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
 &audio_pins {
 	brcm,pins = <>;
 	brcm,function = <>;
-- 
2.34.1

